### PR TITLE
Updated the documentation for FolderBrowserDialog

### PR DESCRIPTION
**Pull request by sas41**
Updated the documentation for FolderBrowserDialog to better reflect the differences between the older and newer versions of .NET

In the newer version of .Net the FolderBrowserDialog does not use the simple Tree View file browser, but uses the modernized File System browser instead, this is more in line with the OpenFileDialog class.

## Summary

Describe your changes here.

Changes the wording around some properties and descriptions to better reflect the differences listed above.

### CODE REVIEW

**Comment by sas41** [comment]
> @sas41 please read the following Contributor License Agreement(CLA). If you agree with the CLA, please reply with the following information.
> 
> ```
> @dotnet-policy-service agree [company="{your company}"]
> ```
> 
> > Options:
> > 
> > * (default - no company specified) I have sole ownership of intellectual property rights to my Submissions and I am not making Submissions in the course of work for my employer.
> > 
> > ```
> > @dotnet-policy-service agree
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 
> > * (when company given) I am making Submissions in the course of work for my employer (or my employer has intellectual property rights in my Submissions by contract or applicable law). I have permission from my employer to make Submissions and enter into this Agreement on behalf of my employer. By signing below, the defined term “You” includes me and my employer.
> > 
> > ```
> > @dotnet-policy-service agree company="Microsoft"
> > ```
> 
> Contributor License Agreement

@dotnet-policy-service agree

**Comment by sas41** [comment]
@dotnet-policy-bot agree

File: xml/System.Windows.Forms/FolderBrowserDialog.xml [hunk 1]
```diff
@@ -47,7 +47,7 @@
       <format type="text/markdown"><![CDATA[  
   
 ## Remarks  
- This class provides a way to prompt the user to browse, create, and eventually select a folder. Use this class when you only want to allow the user to select folders, not files. Browsing of the folders is done through a tree control. Only folders from the file system can be selected; virtual folders cannot.  
+ This class provides a way to prompt the user to browse, create, and eventually select a folder. Use this class when you only want to allow the user to select folders, not files. Browsing of the folders is done through a tree control, from .Net Core 3.1 and onwards, this class uses the modernized file system browser window. Only folders from the file system can be selected; virtual folders cannot.
```
Comment (RussKie) [inline xml/System.Windows.Forms/FolderBrowserDialog.xml:50]:
```suggestion
 This class provides a way to prompt the user to browse, create, and eventually select a folder. Use this class when you only want to allow the user to select folders, not files. Browsing of the folders is done through a tree control, from .NET Core 3.1 and onwards, this class uses the modernized file system browser window. Only folders from the file system can be selected; virtual folders cannot.  
```

File: xml/System.Windows.Forms/FolderBrowserDialog.xml [hunk 2]
```diff
@@ -682,7 +682,7 @@
           <format type="text/markdown"><![CDATA[  
   
 ## Remarks  
- When <xref:System.Windows.Forms.FolderBrowserDialog.ShowNewFolderButton%2A> is `true`, the **New Folder** button is visible, giving the user a chance to create a folder. When the user clicks the **New Folder** button, a new folder is created and the user is prompted to specify the folder name. The selected node in the tree becomes the parent of the new folder. The actual caption of the **New Folder** button can vary depending upon the operating system  
+ When <xref:System.Windows.Forms.FolderBrowserDialog.ShowNewFolderButton%2A> is `true`, the **New Folder** button is visible, giving the user a chance to create a folder. When the user clicks the **New Folder** button, a new folder is created and the user is prompted to specify the folder name. The selected node in the tree becomes the parent of the new folder, in .Net Core 3.1 and later the button is shown regardless of this property and the new folder is created inside the currently opened folder of the modernized file system browser. The actual caption of the **New Folder** button can vary depending upon the operating system
```
Comment (RussKie) [inline xml/System.Windows.Forms/FolderBrowserDialog.xml:685]:
```suggestion
 When <xref:System.Windows.Forms.FolderBrowserDialog.ShowNewFolderButton%2A> is `true`, the **New Folder** button is visible, giving the user a chance to create a folder. When the user clicks the **New Folder** button, a new folder is created and the user is prompted to specify the folder name. The selected node in the tree becomes the parent of the new folder, in .NET Core 3.1 and later the button is shown regardless of this property and the new folder is created inside the currently opened folder of the modernized file system browser. The actual caption of the **New Folder** button can vary depending upon the operating system  
```

**Review by RussKie — COMMENTED** [review]
👍

**Review by RussKie — APPROVED** [review]

**Review by merriemcgaw — APPROVED** [review]
lgtm

**Pull request merged**